tests/shared/file-extensions.spec.js

```diff
@@ -0,0 +1,141 @@
+
+    _.forEach([
+
+      // No extension
+      {
+        file: 'path/to/filename',
+        extension: []
+      },
+
+      // Type: 'archive'
+      {
+        file: 'path/to/filename.zip',
+        extension: [ 'zip' ]
+      },
+      {
+        file: 'path/to/filename.etch',
+        extension: [ 'etch' ]
+      },
+
+      // Type: 'compressed'
+      {
+        file: 'path/to/filename.img.gz',
+        extension: [ 'img', 'gz' ]
+      },
+      {
+        file: 'path/to/filename.img.bz2',
+        extension: [ 'img', 'bz2' ]
+      },
+      {
+        file: 'path/to/filename.img.xz',
+        extension: [ 'img', 'xz' ]
+      },
+
+      // Type: 'image'
+      {
+        file: 'path/to/filename.img',
+        extension: [ 'img' ]
+      },
+      {
+        file: 'path/to/filename.iso',
+        extension: [ 'iso' ]
+      },
+      {
+        file: 'path/to/filename.dsk',
+        extension: [ 'dsk' ]
+      },
+      {
+        file: 'path/to/filename.hddimg',
+        extension: [ 'hddimg' ]
+      },
+      {
+        file: 'path/to/filename.raw',
+        extension: [ 'raw' ]
+      },
+      {
+        file: 'path/to/filename.dmg',
+        extension: [ 'dmg' ]
+      }
+
+    ], (testCase) => {
+      it(`should return ${testCase.extension} for ${testCase.file}`, function() {
+        m.chai.expect(fileExtensions.getFileExtensions(testCase.file)).to.deep.equal(testCase.extension);
+      });
+    });
+
+    it('should always return lowercase extensions', function() {
+      const filePath = 'foo.IMG.gZ';
+      m.chai.expect(fileExtensions.getFileExtensions(filePath)).to.deep.equal([
+        'img',
+        'gz'
+      ]);
+    });
+
+  });
+
+  describe('.getLastFileExtension()', function() {
+
+    it('should return undefined in the file path has no extension', function() {
+      m.chai.expect(fileExtensions.getLastFileExtension('foo')).to.be.undefined;
+    });
+
+    it('should return the extension if there is only one extension', function() {
+      m.chai.expect(fileExtensions.getLastFileExtension('foo.img')).to.equal('img');
+    });
+
+    it('should return the last extension if there two extensions', function() {
+      m.chai.expect(fileExtensions.getLastFileExtension('foo.img.gz')).to.equal('gz');
+    });
+
+    it('should return the last extension if there are three extensions', function() {
+      m.chai.expect(fileExtensions.getLastFileExtension('foo.bar.img.gz')).to.equal('gz');
+    });
+
+  });
+
+  describe('.getPenultimateFileExtension()', function() {
+
+    it('should return undefined in the file path has no extension', function() {
+      m.chai.expect(fileExtensions.getPenultimateFileExtension('foo')).to.be.undefined;
+    });
+
+    it('should return undefined if there is only one extension', function() {
+      m.chai.expect(fileExtensions.getPenultimateFileExtension('foo.img')).to.be.undefined;
+    });
+
+    it('should return the first extension if there are two extensions', function() {
```

Comment:
Doesn't make much difference, but I think for consistency you should use "penultimate" here instead of "first".